pkg/initiailizer_v2/dataset/huggingface.py

```diff
@@ -0,0 +1,42 @@
+import logging
+from urllib.parse import urlparse
+
+import huggingface_hub
+
+import pkg.initiailizer_v2.utils.utils as utils
+
+# TODO (andreyvelich): This should be moved to SDK V2 constants.
+import sdk.python.kubeflow.storage_initializer.constants as constants
+from pkg.initiailizer_v2.dataset.config import HuggingFaceDatasetConfig
+
+logging.basicConfig(
+    format="%(asctime)s %(levelname)-8s [%(filename)s:%(lineno)d] %(message)s",
+    datefmt="%Y-%m-%dT%H:%M:%SZ",
+    level=logging.INFO,
+)
+
+
+class HuggingFace(utils.DatasetProvider):
+
+    def load_config(self):
+        config_dict = utils.get_config_from_env(HuggingFaceDatasetConfig)
+        logging.info(f"Config for HuggingFace dataset initiailizer: {config_dict}")
+        self.config = HuggingFaceDatasetConfig(**config_dict)
+
+    def download_dataset(self):
```

Comment:
Do we have unit tests for these, or would it be taken care in e2e? Downloading models from HF was having issues previously, it would be helpful to have it tested.

Reply:
Nvmd, just realised we have: https://github.com/kubeflow/training-operator/issues/2305